### PR TITLE
[DotNet] Add Bot Authentication SNI sample bot

### PR DESCRIPTION
**Pull request by JhontSouth**
#minor 

## Description
This PR adds a new sample for csharp_dotnetcore that shows how to authenticate the Azure bot using the Subject Name/Issuer from MSAL.

## Proposed Changes
  - Added the new sample _**bot-authentication-sni**_.
  - Added a simple bot conversation to validate the authentication.
  - Added README file with the instructions and information about the sample and how to create a SSL/TSL certificate.

## Testing
This image shows the new _**bot-authentication-sni**_ sample working.
image

### CODE REVIEW

File: samples/csharp_dotnetcore/85.bot-authentication-sni/Bots/AuthCertificateBot.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Logging;
```
Comment (ceciliaavila) [inline samples/csharp_dotnetcore/85.bot-authentication-sni/Bots/AuthCertificateBot.cs]: Dialogs and Logging are not necessary.

File: samples/csharp_dotnetcore/85.bot-authentication-sni/Bots/AuthCertificateBot.cs
Comment (ceciliaavila) [inline samples/csharp_dotnetcore/85.bot-authentication-sni/Bots/AuthCertificateBot.cs:1]: Shouldn't this be called AuthSNIBot ?

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

**Review by sw-joelmut — APPROVED** [review]

File: samples/csharp_dotnetcore/85.bot-authentication-sni/Bots/AuthCertificateBot.cs
Comment (JhontSouth) [reply samples/csharp_dotnetcore/85.bot-authentication-sni/Bots/AuthCertificateBot.cs:1]: Yes, you're right

**Review by ceciliaavila — APPROVED** [review]

**Comment by JhontSouth** [comment]
Promoted to 3917

**Pull request closed without merging**